### PR TITLE
DP-1665: encapsulate movai-service

### PR DESCRIPTION
**Pull request by mariana-dias-alves**
Goal:
- encapsulate python dependencies in debian packaging of movai-service and movai-cli so that it works on focal, jammy and noble

What is done:
- re-work cicd to build 3 packages
- re-work cicd to publish to different ppa's (focal, jammy and noble ppa to be created)
     - https://github.com/MOV-AI/movai-service/actions/runs/12157211677
     - https://github.com/MOV-AI/movai-service-temp/actions/runs/12116587834
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-devel-noble:packages%2Fm%2Fmovai-service%2F0.0.0-9
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-testing-noble:packages%2Fm%2Fmovai-service%2F0.0.0-9
          - https://github.com/MOV-AI/movai-service-temp/releases/tag/0.0.0-9
     - https://github.com/MOV-AI/movai-service-temp/actions/runs/12117706518
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-stable-jammy:packages%2Fm%2Fmovai-cli%2F0.0.0-9

-------------------------

- [x] Make sure you are opening from a **topic/feature/bugfix branch**
- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

[^note]:
    Put an `x` into the [ ] to show you have filled the information.
    The template comes from https://github.com/MOV-AI/.github/blob/master/.github/pull_request_template.md
    You can override it by creating .github/pull_request_template.md  in your own repository

### CODE REVIEW

File: .github/workflows/service-py-deb-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+    inputs:
+      deploy:
+        required: false
+        type: string
+        default: 'false'
+      release:
+        required: false
+        type: string
+        default: 'false'
+      run_black:
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:13]: do we still need those run_black, skip_testing options ?

File: .github/workflows/service-py-deb-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+        default: '["./"]'
+      distributions:
+        required: false
+        type: string
+        default: '["focal", "jammy", "noble"]'
+      install_test:
+        required: false
+        type: string
+        default: 'true'
+      install_test_docker:
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: same here install_test_docker and install_test_zmq are now covered by the makefile

File: .github/workflows/service-py-deb-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,807 @@
...
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: '${{ secrets.auto_commit_password }}'
+          MERGE_MSG: "Branch was auto-updated."
+
+  StaticAnalysis:
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.aws.cloud.mov.ai/devops/py-buildserver:latest
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:78]: I think @duartecoelhomovai recommends not to use latest on py-buildserver, better to mention https://github.com/MOV-AI/ci-agent-docker-py-build-server/releases/tag/v3.0.4

File: .github/workflows/service-py-deb-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,807 @@
...
+    - name: Install build-requirements
+      run: |
+        if [ -f build-apt-requirements.txt ]; then
+          apt update
+          echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+          apt install -y $(grep -vE "^\s*#" build-apt-requirements.txt  | tr "\n" " ")
+        fi
+        python3 -m pip install -r build-requirements.txt
+
+    - name: Run Black
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:99]: please check with @diasdm  which of black, flake8 and pylint needs to be ran now in python CICD

File: .github/workflows/service-py-deb-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,807 @@
...
+
+  Build:
+    needs: [StaticAnalysis]
+    strategy:
+        matrix:
+          pack_path: ${{ fromJSON(inputs.package_paths) }}
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.aws.cloud.mov.ai/devops/py-buildserver:latest
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:146]: use https://github.com/MOV-AI/ci-agent-docker-py-build-server/releases/tag/v3.0.4

File: .github/workflows/service-py-deb-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,807 @@
...
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Install build-requirements
+      run: |
+        python3 -m pip install -r build-requirements.txt
+
+    - name: Create links for package ${{ matrix.pack_path }}
+      run: |
+        ln -f ${{ matrix.pack_path }}/debian/setup.py setup.py
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:166]: now that you fixed the bumpversion, I htink you can get rid of the links ...

File: .github/workflows/service-py-deb-workflow.yml [hunk 7]
```diff
@@ -0,0 +1,807 @@
...
+          PACKAGE_VERSION: ${{ needs.Build.outputs.package_version }}
+        run: |
+          make install_bump_requirements
+          make bump_version
+
+      - name: Find Package details
+        id: vars
+        run: |
+          PACKAGE_VERSION=$(cat .bumpversion.cfg | sed s/' '/''/g | grep 'current_version=' | sed s/'current_version='/''/g)
+          #echo ::set-output name=py_pkg_version::$PACKAGE_VERSION
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: remove commented code

File: .github/workflows/service-py-deb-workflow.yml [hunk 8]
```diff
@@ -0,0 +1,807 @@
...
+          git config --global user.name ${{ secrets.auto_commit_user }}
+          git config --global user.email ${{ secrets.auto_commit_mail }}
+          git add .bumpversion.cfg
+          git commit -m "[skip actions] Automatic Bump of build version"
+
+      - name: Set branch output
+        id: var_branch
+        run: |
+          echo "Branch name is: ${GITHUB_REF#refs/heads/}"
+          #echo ::set-output name=branch::${GITHUB_REF#refs/heads/}
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: remove commented code

File: .github/workflows/service-py-deb-workflow.yml [hunk 9]
```diff
@@ -0,0 +1,807 @@
...
+          remove-dotnet: 'true'
+          remove-codeql: 'true'
+          remove-docker-images: 'true'
+          remove-large-packages: 'true'
+          verbose: 'true'
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: there is no submodules , here

File: .github/workflows/service-py-deb-workflow.yml [hunk 10]
```diff
@@ -0,0 +1,807 @@
...
+        with:
+          pattern: deb_artifacts_${{ matrix.distro }}
+          path: artifacts
+          merge-multiple: true
+
+      - name: Install debian artifacts
+        run: |
+          sudo apt update
+          cd artifacts/
+          find ./ -name "*.deb" -not -name "*src*" -exec sudo apt install -y {} \;
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: make sure to install first the service and then the cli

**Review by AlexFernandes-MOVAI — APPROVED** [review]
some comments on old code, nothing blocking

File: .github/workflows/service-py-deb-workflow.yml (hunk 5 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:146]: This version does not have make command installed: https://github.com/MOV-AI/movai-service/actions/runs/12161966554/job/33918795711#step:6:8

File: .github/workflows/service-py-deb-workflow.yml (hunk 6 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:166]: Attempted locally and did not work: 
```
root@66ddcd7b5df8:/opt/mov.ai# python3 -m build -o dist/movai_service
ERROR Source /opt/mov.ai does not appear to be a Python project: no pyproject.toml or setup.py
```

File: .github/workflows/service-py-deb-workflow.yml (hunk 1 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:13]: for the future

File: .github/workflows/service-py-deb-workflow.yml (hunk 3 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:78]: This version does not have make command installed: https://github.com/MOV-AI/movai-service/actions/runs/12161966554/job/33918795711#step:6:8

File: .github/workflows/service-py-deb-workflow.yml (hunk 4 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:99]: for the future

File: .github/workflows/service-py-deb-workflow.yml (hunk 3 above)
Comment (mariana-dias-alves) [reply .github/workflows/service-py-deb-workflow.yml:78]: needs to be fixed with new image

**Pull request merged**